src/components/Requirements/SubRequirement.vue

```diff
@@ -255,7 +255,13 @@ export default Vue.extend({
         if (!this.isCompleted) {
           slots.push({
             isCompleted: false,
-            courses: generateSubReqIncompleteCourses(allTakenCourseIds, subReqEligibleCourses[0]),
+            courses: generateSubReqIncompleteCourses(
+              allTakenCourseIds,
+              subReqEligibleCourses[0]
+            ).map(course => ({
+              ...course,
+              requirementID: this.subReq.requirement.id,
+            })),
```

Comment:
maybe move the logic to `generateSubReqIncompleteCourses` so no need to repeat below?